sql/analyzer/exec_builder.go

```diff
@@ -180,7 +180,12 @@ func (b *ExecBuilder) buildHashJoin(j *hashJoin, input sql.Schema, children ...s
 	if err != nil {
 		return nil, err
 	}
-	outerAttrs, err := b.buildFilters(j.g.m.scope, j.right.relProps.OutputCols(), expression.Tuple(j.outerAttrs))
+	tmpScope := j.g.m.scope
+	if tmpScope != nil {
+		tmpScope = tmpScope.newScopeInJoin(nil)
```

Comment:
could we have slightly more expressive methods? like `j.g.m.scope.newScopeNoJoin()`?